### PR TITLE
Derive `Clone` along with `Copy` on Rust 1.21

### LINKED ISSUE #934: Derive Clone when deriving Copy
bindgen almost always implements `Clone` explicitly, like this:

```rust
impl Clone for Foo {
    fn clone(&self) -> Self { *self }
}
```

I (accidentally!) ran clippy over some autogenerated code and it expressed disappointment that `Clone` was manually implemented for all sorts of structures on which `Copy` was derived:

```
warning: you are implementing `Clone` explicitly on a `Copy` type
```
And indeed, `Clone` could be derived in every case, at least for my project.

The code that prevents this is [here](https://github.com/rust-lang-nursery/rust-bindgen/blob/dc253cb8ce7f899fa2322d2e3763507448d51553/src/codegen/mod.rs#L1498-L1508).  

My input header is a plain C header (so certainly no templates).  If anything, it almost looks as though that test is exactly backwards, though I expect it's more likely that I simply don't understand what issue is being avoided here...

Anyway, it certainly seems like `Clone` could be derived a lot more often than it currently is.

### PR DESCRIPTION
**Pull request by pepyakin**
Fixes #934

r? @fitzgen or @emilio

### CODE REVIEW

File: src/codegen/mod.rs [hunk 1]
```diff
@@ -1642,16 +1642,21 @@ impl CodeGenerator for CompInfo {
...
-                //     fn clone(&self) -> Self { *self }
-                // doesn't work for templates.
-                //
-                // It's not hard to fix though.
+
+            if ctx.options().rust_features().builtin_clone_impls() {
                 derives.push("Clone");
             } else {
-                needs_clone_impl = true;
+                if used_template_params.is_some() {
```
Comment (fitzgen) [inline src/codegen/mod.rs]: Flatten the nested if/else?

```rust
if ctx.options().rust_features().builtin_clone_impls() || used_template_params.is_some() {
    ...
} else {
    ...
}
```

**Comment by pepyakin** [comment]
@fitzgen done!

**Pull request merged**